code/modules/events/wizard/rpgloot.dm

```diff
@@ -36,28 +19,97 @@
 				upgrade_scroll_chance = max(0,upgrade_scroll_chance-100)
 			upgrade_scroll_chance += 25
 
+	GLOB.rpg_loot_items = TRUE
+
 /obj/item/upgradescroll
-	name = "Item Fortification Scroll"
+	name = "item fortification scroll"
 	desc = "Somehow, this piece of paper can be applied to items to make them \"better\". Apparently there's a risk of losing the item if it's already \"too good\". <i>This all feels so arbitrary...</i>"
 	icon = 'icons/obj/wizard.dmi'
 	icon_state = "scroll"
 	w_class = WEIGHT_CLASS_TINY
 
+	var/upgrade_amount = 1
+	var/can_backfire = TRUE
+	var/one_use = TRUE
+
 /obj/item/upgradescroll/afterattack(obj/item/target, mob/user , proximity)
 	if(!proximity || !istype(target))
 		return
-	var/quality = target.force - initial(target.force)
-	if(quality > 9 && prob((quality - 9)*10))
-		to_chat(user, "<span class='danger'>[target] catches fire!</span>")
-		if(target.resistance_flags & (LAVA_PROOF|FIRE_PROOF))
-			target.resistance_flags &= ~(LAVA_PROOF|FIRE_PROOF)
-			target.resistance_flags |= FLAMMABLE
-		target.fire_act()
+
+	var/datum/rpg_loot/rpg_loot_datum = target.rpg_loot
+	if(!istype(rpg_loot_datum))
+		new rpg_loot_datum(target)
+
+	var/quality = rpg_loot_datum.quality
+
+	if(can_backfire && (quality > 9 && prob((quality - 9)*10)))
+		to_chat(user, "<span class='danger'>[target] violently glows blue for a while, then evaporates.</span>")
+		qdel(target)
```

Comment:
fire_act wasn't doing anything, the code was wrong, and the appropriate code for making things catch fire appropriately even if normally immune to catch fire is ages long.